sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/rows.scala

```diff
@@ -201,7 +201,7 @@ class GenericRow(protected[sql] val values: Array[Any]) extends Row {
 
   override def toSeq: Seq[Any] = values.toSeq
 
-  override def copy(): Row = this
+  override def copy(): Row = new GenericRow(values.clone())
```

Comment:
I'm not sure what's the best solution. Maybe we need some help from more experienced committers here. I think we are agreed that we need a change here, but we differ how the change should look like. 

Anyway, it's not urgent, because the wrong behave should only affect very little developers/users.